mono/metadata/threads.c

```diff
@@ -2314,6 +2314,22 @@ ves_icall_System_Threading_WaitHandle_Wait_internal (gpointer *handles, gint32 n
 
 	MonoW32HandleWaitRet ret;
 
+#ifdef DISABLE_THREADS
+	if (numhandles == 1 && timeout == MONO_INFINITE_WAIT) {
```

Comment:
the `numhandles == 1` seems that is not needed

Reply:
It fails some tests without it. So this is just a hack to detect some deadlocking apps, it doesn't have to catch all cases.

Reply:
I guess the loop computation of signaled below is overkill then

Reply:
Already signalled/owned events can be waited for since the wait will end immediately.